packages/flame/test/components/component_test.dart

```diff
@@ -327,6 +327,81 @@ void main() {
           expect(child.isMounted, true);
         },
       );
+
+      group('lifecycleEventsProcessed', () {
+        testWithFlameGame('waits for unprocessed events', (game) async {
+          await game.ready();
+          final component = _LifecycleComponent();
+          await game.world.add(component);
+          expect(game.hasLifecycleEvents, isTrue);
+
+          Future.delayed(Duration.zero).then((_) => game.update(0));
+          await game.lifecycleEventsProcessed;
+          expect(game.hasLifecycleEvents, isFalse);
+        });
+
+        testWithFlameGame("doesn't block when there are no events",
+            (game) async {
+          await game.ready();
+          expect(game.hasLifecycleEvents, isFalse);
+          await game.lifecycleEventsProcessed;
+          expect(game.hasLifecycleEvents, isFalse);
+        });
+
+        testWithFlameGame('guarantees addition even with heavy onLoad',
+            (game) async {
+          await game.ready();
+          final component = _SlowComponent('heavy', 0.1);
+          final child = _SlowComponent('child', 0.1);
+          await component.add(child);
+          await game.world.add(component);
+          expect(game.world.children, isNot(contains(component)));
+
+          game.lifecycleEventsProcessed.then(
+            expectAsync1((_) {
+              expect(game.world.children, contains(component));
+              expect(component.children, contains(child));
+            }),
+          );
+
+          await game.ready();
+        });
+
+        testWithFlameGame('completes even with dequeued event', (game) async {
+          final parent1 = Component();
+          final parent2 = Component();
+          game.addAll([parent1, parent2]);
+          await game.ready();
+          final component = _SlowComponent('heavy', 0.1);
+          final child = _SlowComponent('child', 0.1);
+          await component.add(child);
+          await parent1.add(component);
+
+          expect(game.lifecycleEventsProcessed, completes);
+
+          await Future.delayed(Duration.zero).then((_) => game.update(0));
+          assert(
+              game.hasLifecycleEvents,
+              'One update should not have been enough '
+              'to add the heavy component');
```

Comment:
Missing trailing comma, is the formatting check not running for tests... :thinking:

Reply:
Addressed